d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -0,0 +1,143 @@
+package com.linkedin.d2.xds;
+
+import com.google.protobuf.Value;
+import com.linkedin.d2.balancer.properties.ClusterProperties;
+import com.linkedin.d2.balancer.properties.ServiceProperties;
+import com.linkedin.d2.balancer.properties.UriProperties;
+import com.linkedin.d2.discovery.event.PropertyEventBus;
+import io.envoyproxy.envoy.config.endpoint.v3.LocalityLbEndpoints;
+import io.envoyproxy.envoy.config.route.v3.Route;
+import io.envoyproxy.envoy.config.route.v3.VirtualHost;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+
+
+public class XdsToD2PropertiesAdaptor
+{
+  private final XdsClient _xdsClient;
+  private PropertyEventBus<UriProperties> _uriEventBus;
+  private PropertyEventBus<ServiceProperties> _serviceEventBus;
+  private PropertyEventBus<ClusterProperties> _clusterEventBus;
+
+  public XdsToD2PropertiesAdaptor(XdsClient xdsClient)
+  {
+    _xdsClient = xdsClient;
+  }
+
+  public static void main(String[] args)
+  {
+    XdsChannelFactory xdsChannelFactory = new XdsChannelFactory();
+    XdsClientImpl xdsClient = new XdsClientImpl(Node.DEFAULT_NODE, xdsChannelFactory.createChannel("localhost:15010"),
```

Comment:
Let's add an option to allow overriding the node here, it will be useful for debugging on Istio side.

Reply:
Sure, added